netket/experimental/operator/_fermion_operator_2nd_jax.py

```diff
@@ -242,14 +246,22 @@ def _apply_terms_scan_bits(x, w, sites, daggers, unroll, process, n_orbitals):
 def apply_terms_scan_bits(
     x, w, sites, daggers, unroll=1, process=True, n_orbitals=None
 ):
-    return _apply_terms_scan_bits(x, w, sites, daggers, unroll, process, n_orbitals)
+    x_res, *res = _apply_terms_scan_bits(
+        x.astype(jnp.uint8),
```

Comment:
should we not put the cast inside the jit context? So it can be fused with other operations by XLA.
At the expense of extra recompilation in some cases..

Reply:
I'ts inside [get_conn_padded_jax](https://github.com/netket/netket/blob/9fea03339f6893a8066941f7da2c7fdd6cb186ce/netket/experimental/operator/_fermion_operator_2nd_jax.py#L399) and [n_conn_jax](https://github.com/netket/netket/blob/9fea03339f6893a8066941f7da2c7fdd6cb186ce/netket/experimental/operator/_fermion_operator_2nd_jax.py#L477C1-L477C1) which are both jitted.

Reply:
ah right. Ok then. all good. Merge away if tests pass.